src/views/types.ts

```diff
@@ -1,3 +1,66 @@
+import { renderArticle, IArticle, INewsState } from '../state/news/types';
+import { renderMeasurment, IMeasurement, IWeatherState } from '../state/weather/types';
+import { IObservable, IObserver } from '../utils/observable/types';
+
 export interface IView {
     render(): void;
 }
+
+export class ViewBase implements IView, IObserver {
+    protected weatherMeasurments: IMeasurement[] = [];
+    protected articles: IArticle[] = [];
+    protected maxWeatherCount: number = 0;
+    protected maxArticleCount: number = 0;
+    protected viewName: string = '';
+    protected lastRenderedPage: string | undefined;
```

Comment:
Тип можно задать иначе:
```suggestion
    protected lastRenderedPage: string?;
```

Reply:
А у меня тогда ошибка, что свойство не проинициализировано. Конструктор создавать для одного поля как то не хочется. Автофикс предлагает ещё написать вот так: `protected lastRenderedPage!: string;`, со словами `Add definite assigment assertion` но мне непонятно, что это значит

Reply:
Восклицательный знак проверяет, что поле не является `null` или `undefined`

А вот так можно?
```suggestion
    protected lastRenderedPage?: string;
```

Reply:
Ой, да, так даёт, спасибо! :)